impl/src/main/java/org/ehcache/config/builders/ConfigurationBuilder.java

```diff
@@ -78,7 +78,9 @@ public Configuration build() {
 
   ConfigurationBuilder addCache(String alias, CacheConfiguration<?, ?> config) {
     Map<String, CacheConfiguration<?, ?>> newCaches = new HashMap<String, CacheConfiguration<?, ?>>(caches);
-    newCaches.put(alias, config);
+    if(newCaches.put(alias, config) != null) {
```

Comment:
I think the builders reusing can be refactored to prevent adding an existing alias. In the issue will be super rare. So I think adding `withCacheOverride` is overkill